app/assets/javascripts/admin/addon/components/file-size-input.js

```diff
@@ -0,0 +1,132 @@
+import Component from "@ember/component";
+import { action, computed } from "@ember/object";
+import { classNames } from "@ember-decorators/component";
+import I18n from "discourse-i18n";
+
+const ALLOWED_KEYS = [
+  "Enter",
+  "Backspace",
+  "Tab",
+  "Delete",
+  "ArrowLeft",
+  "ArrowUp",
+  "ArrowRight",
+  "ArrowDown",
+  "0",
+  "1",
+  "2",
+  "3",
+  "4",
+  "5",
+  "6",
+  "7",
+  "8",
+  "9",
+];
+
+/**
+  An input field for a file size.
+
+**/
+@classNames("file-size-picker")
+export default class FileSizeInput extends Component {
```

Comment:
It's recommended to use glimmer/gjs for new components

Reply:
Well now that I have it working, let me try and convert it.